src/instrumenter/instrument.ts

```diff
@@ -978,15 +1017,35 @@ function insertVarsStruct(
 }
 
 function getCheckStateInvsFuncs(contract: ContractDefinition): FunctionDefinition {
-    return single(contract.vFunctions.filter((fn) => fn.name === CHECK_STATE_INVS_FUN));
+    return single(
+        contract.vFunctions.filter(
+            (fn) =>
+                fn.name === CHECK_STATE_INVS_FUN || fn.name.slice(0, -2) === CHECK_STATE_INVS_FUN
+        )
+    );
 }
 
 function isPublic(fn: FunctionDefinition): boolean {
     return [FunctionVisibility.Default, FunctionVisibility.Public].includes(fn.visibility);
 }
 
-function getInternalCheckInvsFun(contract: ContractDefinition): string {
-    return `__scribble_${contract.name}_check_state_invariants_internal`;
+function getInternalCheckInvsFun(
+    contract: ContractDefinition,
+    contractInvariantsData: ContractInvariantsData | undefined
+) {
+    if (contractInvariantsData) {
+        return contractInvariantsData.internalInvariantFunction;
+    }
+
+    const allNames = getAllNames(contract);
+    let funcName = `__scribble_${contract.name}_check_state_invariants_internal`;
+
+    if (!allNames.has(funcName)) {
```

Comment:
This pattern appears in several places:

`if (!someNames.has(name)) {
   return name;
 }
 
 name = getNewVarName(names, "${name}_");
 `
 
 Perhaps we can roll that `if` inside of `getNewVarName`, such that `getNewVarName(names, str)` returns `str` when its not inside `names`, and otherwise returns `${str}_[0-9]*`